code/game/mecha/mech_fabricator.dm

```diff
@@ -269,7 +269,7 @@
 	updateUsrDialog()
 	sleep(30) //only sleep if called by user
 
-	for(var/obj/machinery/computer/rdconsole/RDC in area_contents(get_area(src)))
+	for(var/obj/machinery/computer/rdconsole/RDC in oview(5,src))

```

Comment:
I forget if this is how it works but I think this (or something like this) used to run on oview and it lead to a few baffled player cases where they would try to sync machines but fail because the lights had gone out.

It's probably ok to suffer an orange() check when it's usually a once per round event.